### PR TITLE
SKW-3961: futurize conduce api

### PR DESCRIPTION
**Pull request by konvulse**
Applies stage 1 and stage 2 as outlined in the quickstart guide here:

https://python-future.org/automatic_conversion.html

The only deviation is in some of the imports (see conduce.py for example).  Imports are to follow the convention `from . import config` but this convention produced errors in both Python 2 and Python 3. I think the project would need restructuring to make it work.

Hashtags #python3 #python-future.

### CODE REVIEW

**Comment by konvulse** [comment]
Additional changes so far:
argparse subcommands
pickle binary file operations

**Comment by konvulse** [comment]
Successfully ran create_world_cities.py

**Comment by konvulse** [comment]
After these changes python conduce/conduce.py cannot be run directly in a dev env.  ./conduce.py has been added for that purpose.  conduce/conduce.py has been renamed conduce/cli.py.

**Comment by konvulse** [comment]
It occurs to me that the instructions in the readme may not work anymore.  Specifically: 
```
pip install git+git://github.com/ConduceInc/conduce-python-api
```

File: conduce/ingest.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from . import api
+from . import util
+
+
+def get_dataset_id(dataset_name, **kwargs):
```
Comment (konvulse) [inline conduce/ingest.py]: I don't really care for this function in a file called `ingest.py`.  It should either move to `api.py` or this filename should change.  I like the latter, but I can't think of a good name at this time.

Comment (konvulse) [reply conduce/ingest.py]: I suppose this function could be removed, but it provides a nice interface.  If `find_resource` had an option to return only the first result and not an array this function would not be necessary.  `find_resource(..., first=True).get('id')` would be equivalent

**Comment by konvulse** [comment]
Ready for thorough review.

File: README.md [hunk 2]
```diff
@@ -1,64 +1,70 @@
...
-`api.py` is the majority of the implementation.  `conduce.py` is a command line utility for invoking the API.  It also serves as a representative implementation of the library.
+`conduce/api.py` is the majority of the implementation. `conduce/cli.py` is a command line utility for invoking the API. It also serves as a representative implementation of the library.
 
 # Prerequisites
 
 Conduce Python API requires that you have installed the following:
 
-- Python 2.7: https://www.python.org/downloads/
-- pip: https://pip.pypa.io/en/stable/installing/
+* Python 2.7: https://www.python.org/downloads/
```
Comment (jrmConduce) [inline README.md]: So is this expected to work in both python 2 and python 3 currently?

**Comment by jrmConduce** [comment]
General question: I know that we typically use the latest from git in requirements.txt wherever this library is used, and I know there are some breaking changes here.  Are there any outstanding libraries that need to be updated before this change could be merged?

File: README.md (hunk 2 above)
Comment (konvulse) [reply README.md]: Actually, yes.  It should work with both.  I overlooked the prerequisites.

**Comment by konvulse** [comment]
> General question: I know that we typically use the latest from git in requirements.txt wherever this library is used, and I know there are some breaking changes here. Are there any outstanding libraries that need to be updated before this change could be merged?

Yes, there were changes made to the DHL repositories to stop using functions that were moved to ingest.py.  I'm not sure if they have been merged yet.

**Comment by konvulse** [comment]
DHL PRs have been merged.  We should be good to go here.

**Pull request merged**